content/docs/command-reference/push.md

```diff
@@ -18,55 +18,48 @@ positional arguments:
-[metrics](/doc/command-reference/metrics), etc.)
-[remotely](/doc/command-reference/remote) are the two most common use cases for
-these commands.
+[metrics](/doc/command-reference/metrics), etc.) remotely are the most common
+use cases for these commands.
 
-The `dvc push` command allows us to upload data to remote storage. It doesn't
-save any changes to the code, `dvc.yaml`, or `.dvc` files (those should be saved
-with `git commit` and `git push`).
+The `dvc push` command allows us to upload data to
+[remote storage](/doc/command-reference/remote). It doesn't save any changes to
+the code, `dvc.yaml`, or `.dvc` files (that should be saved with `git commit`
+and `git push`).
 
-💡 For convenience, a Git hook is available to automate running `dvc push` after
-`git push`. See `dvc install` for more details.
-
-Under the hood a few actions are taken:
+The default remote is used (see `dvc config core.remote`) unless the `--remote`
+option is used. See `dvc remote` for more information on how to configure a
+remote.
 
-- The push command by default uses all stages (in `dvc.yaml` and `dvc.lock`) and
-  `.dvc` files in the <abbr>workspace</abbr>. The command options will either
-  limit or expand the set of stages or `.dvc` files to consult.
+Without arguments, it uploads all files and directories missing from remote
+storage, found as <abbr>outputs</abbr> in the stages (in `dvc.lock`) or `.dvc`
+files present in the workspace (the `--all-branches` and `--all-tags` enable
+using multiple workspace versions).
 
-- For each <abbr>output</abbr> referenced in every selected stage or `.dvc`
-  file, DVC finds a corresponding file or directory in the <abbr>cache</abbr>.
-  DVC then checks whether it exists in the remote. From this, DVC gathers a list
-  of files missing from the remote storage.
+The `targets` given to this command (if any) limit what to push. It accepts
+paths to tracked files or directories (including paths inside tracked
+directories), `.dvc` files, or stage names (found in `dvc.yaml`).
 
-- Upload the cache files missing from remote storage, if any, to the remote.
+💡 For convenience, a Git hook is available to automate running `dvc push` after
+`git push`. See `dvc install` for more details.
 
-The DVC `push` command always works with a remote storage, and it is an error if
-none are specified on the command line nor in the configuration. The default
-remote is used (see `dvc config core.remote`) unless the `--remote` option is
-used. See `dvc remote` for more information on how to configure a remote.
+Under the hood, a few actions are taken:
 
-With no arguments, just `dvc push` or `dvc push --remote REMOTE`, it uploads
-only the files (or directories) that are new in the local repository to remote
-storage. It will not upload files associated with earlier commits in the
-<abbr>repository</abbr> (if using Git), nor will it upload files that have not
-changed.
+- The push command checks the appropriate `dvc.lock` and `.dvc` files in the
```

Comment:
dvc.yaml?

Reply:
It's a command-reference, but I don't think this information should be here (perhaps on Dvc files?).

Also, @jorgeorpinel, `dvc.yaml` is a _discovery_ file which dvc reads to find outputs/dependencies, etc. and `dvc.lock` is a file to correlate those outputs/deps with the exact version through the checksums. `dvc.lock` is never used if there's no corresponding deps/outs/stages in dvc.yaml. So, either both dvc.yaml and dvc.lock are read together or we could just say dvc.yaml is read (and consider dvc.lock as an implementation detail).

Reply:
> It's a command-reference, but I don't think this information should be here 

could you clarify please what information exactly do you mean?

> So, either both dvc.yaml and dvc.lock are read together or we could just say dvc.yaml is read (and consider dvc.lock as an implementation detail) ..

we come back to a single term "DVC files" w/o specifying details with some tooltip that mentions that DVC files == dvc.yaml + dvc.lock + .dvc . All of those files play their role here. We can expand later and bit explicit if needed where it is needed.

Reply:
dvc.yaml is mentioned just before: "`targets` ... limit what to pull. It accepts ... and stage names (found in `dvc.yaml`)". Then, when we say "push checks the appropriate `dvc.lock` files", by "appropriate" we mean corresponding to what was found in dvc.yaml files. But let me see if I can clarify this further... ⌛ 

> just say dvc.yaml is read (and consider dvc.lock as an implementation detail

This is not a bad idea... We do go over implementation details in cmd refs though, in fact they'r probably the most low-level docs we have. But we should prob. hide such details in expandable sections. Just not sure dvc.lock should be considered a low-level detail, it's pretty visible to the user.

> we come back to a single term "DVC files" w/o specifying details with some tooltip that mentions that DVC files

Yes, let's try to merge this one as best as possible though, and follow up on this in #1663 ?

Reply:
> See also iterative/dvc/issues/4393

Reply:
> let me see if I can clarify this further... ⌛

Turn this whole bullet list into a more simple paragraph in https://github.com/iterative/dvc.org/pull/1602#pullrequestreview-465516419. PTAL